### PR TITLE
open notification list when tapping push notifications

### LINKED ISSUE #442: Push Notification Deep Links
We should support deep linking into the app from push notifications. Right now when you tap a notification it only opens the app to the default screen.

For example if you tap a transfer notification it should open the notification tab so you see the new transfer request.

### PR DESCRIPTION
**Pull request by tyler-whitman**
closes #442 

This makes it so that when you tap a push notification the app opens to the notifications screen after unlocking. This way when users get a notification and tap it they will see the new notification after opening the app.

### CODE REVIEW

File: src/store/notifications/notificationSlice.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import { createSlice } from '@reduxjs/toolkit'
+import { OSNotification } from 'react-native-onesignal'
+
+export type NotificationState = {
+  notification?: OSNotification
```
Comment (matthewcarlreetz) [inline src/store/notifications/notificationSlice.ts:5]: I think we should move the notifications in `src/store/account/accountSlice.ts` over to here.

**Review by matthewcarlreetz — CHANGES_REQUESTED** [review]

File: src/features/hotspots/root/HotspotsTicker.tsx [hunk 2]
```diff
@@ -58,7 +58,7 @@ const HotspotsTicker = ({ ...boxProps }: Props) => {
   const text = useMemo(() => {
     const formattedHotspotCount = hotspotCount?.toLocaleString(locale) || 0
     const oraclePrice = currentOraclePrice?.price || 0
-    const formattedBlockTime = blockTime.toLocaleString(locale, {
+    const formattedBlockTime = blockTime?.toLocaleString(locale, {
```
[the remark this answers is not in the document]
Comment (tyler-whitman) [reply src/features/hotspots/root/HotspotsTicker.tsx:61]: Ok let me see, I was fixing some state bug here with the types

File: src/navigation/main/MainTabNavigator.tsx [hunk 3]
```diff
@@ -14,29 +14,41 @@ import WalletNavigator from '../../features/wallet/WalletNavigator'
...
 import appSlice from '../../store/user/appSlice'
 import NotificationsScreen from '../../features/notifications/NotificationsScreen'
+import notificationSlice from '../../store/notifications/notificationSlice'
 
 const MainTab = createBottomTabNavigator()
 
 const MainTabs = () => {
   const { white, grayLight } = useColors()
-  const navigation = useNavigation<RootNavigationProp>()
+  const navigation = useNavigation()
```
[the remark this answers is not in the document]
Comment (tyler-whitman) [reply src/navigation/main/MainTabNavigator.tsx:23]: Ok I was trying to figure out what to do there, is there any problem with creating two navigators on one screen?

File: src/store/notifications/notificationSlice.ts (hunk 1 above)
Comment (tyler-whitman) [reply src/store/notifications/notificationSlice.ts:5]: Ah ok I can, maybe I call this new one pushNotifcation?

File: src/navigation/main/MainTabNavigator.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (matthewcarlreetz) [reply src/navigation/main/MainTabNavigator.tsx:23]: It should be ok. MainTabNav is nested inside of RootNat.

File: src/store/notifications/notificationSlice.ts (hunk 1 above)
Comment (matthewcarlreetz) [reply src/store/notifications/notificationSlice.ts:5]: that sounds good.

File: src/navigation/main/MainTabNavigator.tsx [hunk 4]
```diff
@@ -20,34 +25,35 @@ const MainTab = createBottomTabNavigator()
...
-    (state: RootState) => state.notifications.notification,
+  const pushNotification = useSelector(
+    (state: RootState) => state.notifications.pushNotification,
   )
   const dispatch = useAppDispatch()
 
   useEffect(() => {
     if (!isLocked) return
-    navigation.navigate('LockScreen', { requestType: 'unlock', lock: true })
+    navigation.push('LockScreen', { requestType: 'unlock', lock: true })
```
Comment (matthewcarlreetz) [inline src/navigation/main/MainTabNavigator.tsx]: Just curious why these were changed from navigate to push?

**Review by matthewcarlreetz — APPROVED** [review]
💯

File: src/navigation/main/MainTabNavigator.tsx (hunk 4 above)
Comment (tyler-whitman) [reply src/navigation/main/MainTabNavigator.tsx]: Thats what they were originally, I was actually unsure on the difference between the two but I decided to put them back to what they were previously.

Comment (tyler-whitman) [reply src/navigation/main/MainTabNavigator.tsx]: Ah looks like the only difference is push allows the component to be pushed on the stack multiple times. In the case they have the same effect

**Review by allenan — APPROVED** [review]
nice this looks good, the one thing i mentioned was to see if we can trigger the new notifications circle icon on the nav bar to show up when a push notification comes through. definitely like that we're not polling notifications based on blocks anymore 👍

**Pull request merged**